crates/sui-network/src/discovery/server.rs

```diff
@@ -27,7 +31,32 @@ impl Discovery for Server {
             .our_info
             .clone()
             .ok_or_else(|| anemo::rpc::Status::internal("own_info has not been initialized yet"))?;
-        let known_peers = state.known_peers.values().cloned().collect();
+
+        // prefer returning peers that we are connected to as they are known-good
+        let mut known_peers = state
+            .connected_peers
+            .keys()
+            .filter_map(|peer_id| state.known_peers.get(peer_id))
+            .map(|info| (info.peer_id, info))
+            .take(MAX_PEERS_TO_SEND)
+            .collect::<HashMap<_, _>>();
+
+        if known_peers.len() <= MAX_PEERS_TO_SEND {
+            // Fill the remaining space with other peers, randomly sampling at most MAX_PEERS_TO_SEND
+            for info in state
+                .known_peers
+                .values()
+                .choose_multiple(&mut rand::thread_rng(), MAX_PEERS_TO_SEND)
```

Comment:
choose_multiple docs say
"Although the elements are selected randomly, the order of elements in the buffer is neither stable nor fully random. If random ordering is desired, shuffle the result."

Since you are potentially breaking before all the chosen elements are used, does this mean you need to collect them into a vector and shuffle first before merging?

Reply:
No elements are selected randomly its the order of the selected elements that isn't necessarily stable or random